bigchaindb_driver/offchain.py

```diff
@@ -0,0 +1,229 @@
+    fulfilled.
+
+    Args:
+        owners_before (:obj:`list` | :obj:`tuple` | :obj:`str`): One
+            or more public keys representing the original owner(s) of
+            the asset being created.
+        owners_after (:obj:`list` | :obj:`tuple` | :obj:`str`, optional):
+            One or more public keys representing the new owner(s) of the
+            asset being created. Defaults to ``None``.
+        asset (:obj:`dict`, optional): The asset being created. Defaults
+            to ``None``.
+        metadata (:obj:`dict`, optional): Metadata associated with the
+            transaction. Defaults to ``None``.
+
+    Returns:
+        dict: The prepared ``"CREATE"`` transaction.
+
+    """
+    if not isinstance(owners_before, (list, tuple)):
+        owners_before = [owners_before]
+
+    # NOTE: Needed for the time being. See
+    # https://github.com/bigchaindb/bigchaindb/issues/797
+    if isinstance(owners_before, tuple):
+        owners_before = list(owners_before)
+
+    if not owners_after:
+        owners_after = owners_before
+
+    asset = _normalize_asset(asset)
+
+    transaction = Transaction.create(
+        owners_before,
+        owners_after,
+        metadata=metadata,
+        asset=asset,
+    )
+    return transaction.to_dict()
+
+
+def prepare_transfer_transaction(*,
+                                 inputs,
+                                 owners_after,
+                                 asset,
+                                 metadata=None):
+    """
+    Prepares a ``"TRANSFER"`` transaction payload, ready to be
+    fulfilled.
+
+    Args:
+        inputs (:obj:`dict` | :obj:`list` | :obj:`tuple`): One or more
+            inputs holding the condition(s) that this transaction
+            intents to fulfill. Each input is expected to be a
+            :obj:`dict`.
+        owners_after (:obj:`str` | :obj:`list` | :obj:`tuple`): One or
+            more public keys representing the new owner(s) of the asset
+            being transferred.
+        asset (:obj:`dict`): The asset being transferred.
+        metadata (:obj:`dict`): Metadata associated with the
+            transaction. Defaults to ``None``.
+
+    Returns:
+        dict: The prepared ``"TRANSFER"`` transaction.
+
+    """
+    if not isinstance(inputs, (list, tuple)):
+        inputs = (inputs, )
+    if not isinstance(owners_after, (list, tuple)):
+        owners_after = [owners_after]
+
+    fulfillments = [
+        BdbFulfillment(Fulfillment.from_dict(input_['fulfillment']),
+                       input_['owners_before'],
+                       tx_input=TransactionLink(**input_['input']))
+        for input_ in inputs
+    ]
+
+    asset = _normalize_asset(asset, is_transfer=True)
+
+    transaction = Transaction.transfer(
+        fulfillments,
+        owners_after,
+        asset,
+        metadata=metadata,
+    )
+    return transaction.to_dict()
+
+
+def fulfill_transaction(transaction, *, private_keys):
```

Comment:
As discussed offline, the reason for doing so was that this will allow us to introduce the support for hashlocks and other cryptoconditions goodies in a smooth way.

Hashlocks won't require private keys, and therefore we can add a default value of `None` without breaking current users code.